GitUI/Editor/FileViewer.cs

```diff
@@ -467,19 +478,35 @@ public void ViewStagingPatch(Patch patch)
 
         public void ViewPatch([CanBeNull] Patch patch)
         {
-            string text = patch != null ? patch.Text : "";
-            ViewPatch(text);
+            if (patch == null)
+            {
+                ViewNoPatch();
+            }
+            else
+            {
+                ViewPatch(patch.Text, GitUIExtensions.OpenWithDifftoolNotImplemented /* not supported here, but globally in FormCommit */);
+            }
+        }
+
+        public void ViewNoPatch()
+        {
+            ViewPatch(string.Empty, GitUIExtensions.CannotOpenWithDifftool);
         }
 
-        public void ViewPatch([NotNull] string text)
+        public void ViewPatch([NotNull] string text, [CanBeNull] Action openWithDifftool)
         {
             ResetForDiff();
-            _internalFileViewer.SetText(text, isDiff: true);
+            _internalFileViewer.SetText(text, isDiff: true, openWithDifftool);
             TextLoaded?.Invoke(this, null);
             RestoreCurrentScrollPos();
         }
 
-        public Task ViewPatchAsync(Func<string> loadPatchText)
+        private void ViewPatch(Tuple<string, Action> text_openWithDifftool)
+        {
+            ViewPatch(text_openWithDifftool.Item1, text_openWithDifftool.Item2);
+        }
```

Comment:
Why is this method needed and why calls in `ViewCurrentChanges` can't be replaced with calls to `ViewPatch(string, Action)` method?